apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ts

```diff
@@ -123,40 +125,135 @@ router.post(
         );
       }
 
+      const body = z
+        .object({
+          __action: z.string(),
+          __csrf_token: z.string(),
+          orig_hash: z.string(),
+          qid: z.string(),
```

Comment:
We could eliminate the check for the presence of a QID above and shift the format check for the QID to below this body parsing step.